src/main/java/io/supertokens/oauth/exceptions/OAuthClientRegisterException.java

```diff
@@ -0,0 +1,29 @@
+/*
+ *    Copyright (c) 2024, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ *    This software is licensed under the Apache License, Version 2.0 (the
+ *    "License") as published by the Apache Software Foundation.
+ *
+ *    You may not use this file except in compliance with the License. You may
+ *    obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *    License for the specific language governing permissions and limitations
+ *    under the License.
+ */
+
+package io.supertokens.oauth.exceptions;
+
+import java.io.Serial;
+
+public class OAuthClientRegisterException extends OAuthException{
```

Comment:
what during the registration process has gone wrong for this exception to be thrown? The name has to indicate that clearly

Reply:
This is an exception to handle errors returned by hydra. I can't tell at this point what went wrong or what could go wrong. This name indicates that the error happened while we were using the client registration API of hydra, but the concrete error is in the `error` and `errorDescription` fields.

Reply:
renamed